test/unit/functions/noSelfOption.spec.ts

```diff
@@ -0,0 +1,52 @@
+import * as util from "../../util";
+
+test.each([["foo: string, bar: string", `this: void, foo: string, bar: string`]])(
+    'enables noSelfInFile behaviour for functions ("%s" equivalent to "%s")',
+    (expectedParameters, assignmentParameters) => {
+        util.testFunction`
+            function fooBar(${expectedParameters}) {}
+
+            const test: (${assignmentParameters}) => void = fooBar;
```

Comment:
Since function are assignable to types with fewer parameters you can just use `this: void`. Though there's probably no much reason to add any parameters to these functions anyway, I think it's better to keep test code as simple as possible